Add an action for retrieving meeting times from VRMS nightly

cnk opened the pull request:
Fixes #550 

Works in my actions-test repo: https://github.com/cnk/actions-test/actions/runs/120879511

cnk committed:
Add an action for retrieving meeting times from VRMS nightly

KianBadie commented:
@cnk could you please try testing the github action on your website branch? I checked out a branch and pulled your code changes. I then changed the github action to fire on pushes for testing. 

The first time I ran the code, The "build" job failed because of this error
```
[cnk-vrms-data c1171ae] Update contributor and language data
 Author: GitHub Actions Bot <hackforla-bot@hackforla.org>
 1 file changed, 1 insertion(+)
 create mode 100644 _data/vrms_data.json
To https://github.com/KianBadie/website
 ! [rejected]        cnk-vrms-data -> cnk-vrms-data (fetch first)
error: failed to push some refs to 'https://github.com/KianBadie/website'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

I deleted the branch because I thought it was weird that it worked on your test repo and not on my repo. When I tried again on a new branch, I got the same error. ***But*** this time it was for the vrms_data job. 

My theory is that the error comes up because each job is trying to commit code, but the github-data job doesn't have knowledge of the schedule data commit if that job finishes first, and the vrms_data job doesn't have knowledge of the github-data commit if that job finishes first. But that wouldn't explain why it worked on your repository. 

Anyways, that's why I am asking if you could please simulate it on your website repo, that way I can see if it is just on my end.

***Update:***
Interesting thing just happened, so earlier when the "build" job succeeded but the "vrms_data" job failed, I pulled from the branch, got the updates to github-data.json from the "build" job, ran the github action again through a push, and now this time ***both*** jobs succeeded. I will look more into this.

***Update 2:***
I tried again. The first attempt at testing the vrms_data job failed for the same error, so I pulled (only thing that was new was 20ish lines in github-data.json). Second attempt the build job failed (I can't remember if I pulled again and got changes from _data/vrms_data.json). 3rd attempt both jobs succeeded.

On another go, it only took 2 tries. This time with the "build" job failing first. Once I pulled the data (only thing was _data/vrms_data.json) and pushed again it worked. I think my initial theory might be true. If so, maybe a solution could by to combine the 2 jobs into one. That way there would only be one commit?

cnk committed:
Merge branch 'hackforla-gh-pages' into vrms-data

cnk committed:
Renamed main.yml to github-data.yml and added a vrms-data.yml to get meeting data from VRMS

@KianBadie is right. Having these 2 scripts run in the same job will cause commit conflicts if they both have changes.
Split them into their own actions and run them at different times so they will never conflict.

cnk committed:
Fix syntax error in vrms-data.yml (missing jobs keyword)

ExperimentsInHonesty commented:
@KianBadie ready for you to review again.

KianBadie commented:
I tested this by changing the vrms data job to fire on push. I then pulled the data that was added and then changed the github data job to fire on push. Both the files got added. 

I'm assuming when the github job will simulate the pull I did because it will be checking out the project repository an hour after the vrms job commits it's work. 

I am going to go ahead and merge this today because it has a small consequence if it does not work (files won't get committed, but old data will still be there). Rather than testing it overnight tonight by using the regular scheduled times of each job. That way, we can see the results of this asap. Also, I have a feeling that from my light tests and from how the code looks, it will work fine.
